BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java

```diff
@@ -111,6 +116,219 @@ public boolean onOptionsItemSelected(MenuItem item) {
     return super.onOptionsItemSelected(item);
   }
 
+  private void checkEmptyLibrary() {
+    //if no shelf or no books
+
+    dbHelper = new DatabaseHelper(getContext());
+    bookDao = new BookDao(dbHelper);
+
+    if (libraryModel.getCurrentLibraryList().isEmpty() || bookDao.findAllBooks().isEmpty()) {
+      AlertDialog.Builder alertDialogEmptyLib = new AlertDialog.Builder(getContext());
+      alertDialogEmptyLib.setTitle(R.string.empty_library);
+      alertDialogEmptyLib.setMessage(R.string.empty_library_description);
+      alertDialogEmptyLib.setPositiveButton(R.string.ok,
+          new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+            }
+          });
+      alertDialogEmptyLib.create().show();
+
+    } else {
+      checkStoragePermission();
+    }
+  }
+
+  private void checkStoragePermission() {
+    if (ContextCompat.checkSelfPermission(getContext(),
+        Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+      requestStoragePermission();
+    } else {
+      //if the user has already allowed access to device external storage
+      bibExportLibrary();
+    }
+  }
+
+  private void requestStoragePermission() {
+    if (ActivityCompat.shouldShowRequestPermissionRationale(getActivity(),
+        Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+      showRequestPermissionDialog();
+    } else {
+      requestPermissions(new String[] {Manifest.permission.WRITE_EXTERNAL_STORAGE},
+          STORAGE_PERMISSION_CODE);
+    }
+  }
+
+  private void showRequestPermissionDialog() {
```

Comment:
Duplicate code

Reply:
This is a standard method used for relative permissions requests. Because I ask for the same permissions, I give the same Information to the user. There is why there are no changes. What do you mean by flag? A variable from type boolean?

Reply:
Resolved in the discussion.